### ISSUE TITLE
Enabling billing causes 500 if user-provided services exist

### ISSUE THREAD

**Issue by drnic** [cloudfoundry/cloud_controller_ng#325]
See https://groups.google.com/a/cloudfoundry.org/forum/#!topic/vcap-dev/_FHO8K3FfXQ

When I created a fresh organization and enabled billing it worked. When the organization already had a user-provided service it failed with a 500.

Snippet from logs:

```
"Request failed: 500: {\"code\"=>10001, \"description\"=>\"undefined method `service_plan' for #<VCAP::CloudController::UserProvidedServiceInstance:0x007f923ab6b3a8>\"
```

More here:

```
{"timestamp":1421945433.8165245,"message":"Request failed: 500: {\"code\"=>10001, \"description\"=>\"undefined method `service_plan' for #<VCAP::CloudControll
er::UserProvidedServiceInstance:0x007f923b2f4688>\", \"error_code\"=>\"CF-NoMethodError\", \"backtrace\"=>[
\"/var/vcap/packages/cloud_controller_ng/cloud_cont
roller_ng/app/models/services/service_create_event.rb:37:in 
`create_from_service_instance'\", 
\"/var/vcap/packages/cloud_controller_ng/cloud_controller_ng/app
/models/runtime/organization.rb:109:in `block in after_save'\", 
\"/var/vcap/packages/cloud_controller_ng/cloud_controller_ng/app/models/runtime/organization.r
b:108:in `each'\", 
\"/var/vcap/packages/cloud_controller_ng/cloud_controller_ng/app/models/runtime/organization.rb:108:in `after_save'\", 
\"/var/vcap/packages
/cloud_controller_ng/cloud_controller_ng/vendor/bundle/ruby/2.1.0/gems/sequel-4.15.0/lib/sequel/model/base.rb:1864:in `block in _save'\", 
```

**Comment by drnic** [comment]
I guess we need return immediately if its a user-provided services https://github.com/cloudfoundry/cloud_controller_ng/blob/master/app/models/services/service_create_event.rb#L37

**Comment by shalako** [comment]
Hi @drnic, we fixed this. Should be included with the next final release.

https://gist.github.com/shalako/783f4e8425b78d074a77

**Comment by shalako** [comment]
@drnic Just learned more about this in compiling release notes for v198. I did not recognize that the `billing_enabled` field is deprecated (though this is documented in the API docs for /v2/organizations). As you probably know and I did not catch, /v2/billing_events is also deprecated (these endpoints are undocumented). 

Had I understood this we may not have fixed this issue. Regardless, I recommend ceasing use of /v2/billing_events, in favor of /v2/app_usage_events and /v2//v2/service_usage_events.
